### PR TITLE
Fix CI: testPostRequestFormat::TestCustomRestClient

### LINKED ISSUE #2188: [Failed CI Test] testPostRequestFormat(org.apache.helix.rest.client.TestCustomRestClient)
This issue is created for tracking unstable test: testPostRequestFormat(org.apache.helix.rest.client.TestCustomRestClient)

### LINKED ISSUE #2402: [Failed CI Test] testPostRequestFormat(org.apache.helix.rest.client.TestCustomRestClient)
This issue is created for tracking unstable test: testPostRequestFormat(org.apache.helix.rest.client.TestCustomRestClient)

### PR DESCRIPTION
**Pull request by desaikomal**
### Issues

- [x] My PR addresses the following Helix issues and references them in the PR description:
Fixes #2402 and Fixes #2188 

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
The method testPostRequestFormat tries to connect to an external site: http://httpbin.org which is Echo server which will reply back with whatever has been sent.
But the end-point sometimes gives out 504, ie. gateway timeout.

This makes the test unstable. So add the check to see if status is OK, then only validate the rest of the fields.

### Tests

- [ ] The following tests are written for this issue:

- The following is the result of the "mvn test" command on the appropriate module:

mvn test output:
[ERROR] Failures:
[ERROR]   TestNoThrottleDisabledPartitions.testDisablingTopStateReplicaByDisablingInstance:98 expected:<false> but was:<true>
[ERROR]   TestNoThrottleDisabledPartitions.testNoThrottleOnDisabledInstance:231->setupEnvironment:317->setupCluster:436 » ZkClient
[ERROR]   TestP2PNoDuplicatedMessage.testP2PStateTransitionEnabled:180 expected:<true> but was:<false>
[ERROR]   TestDelayedAutoRebalance.testDisableDelayRebalanceInResource:169->validateDelayedMovements:310 expected:<true> but was:<false> 
[ERROR]   TestResourceThreadpoolSize.testBatchMessageThreadPoolSize:206 expected:<true> but was:<false>
[ERROR] Tests run: 1331, Failures: 5, Errors: 0, Skipped: 2

### Changes that Break Backward Compatibility (Optional)

- My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:

(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-rest/src/test/java/org/apache/helix/rest/client/TestCustomRestClient.java [hunk 1]
```diff
@@ -158,11 +158,13 @@ public void testPostRequestFormat() throws IOException {
...
     CustomRestClientImpl customRestClient = new CustomRestClientImpl(HttpClients.createDefault());
     HttpResponse response = customRestClient.post(echoServer, Collections.emptyMap());
-    JsonNode json = customRestClient.getJsonObject(response);
+    if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+      JsonNode json = customRestClient.getJsonObject(response);
 
-    Assert.assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_OK);
-    Assert.assertEquals(json.get("headers").get("Accept").asText(), "application/json");
-    Assert.assertEquals(json.get("data").asText(), "{}");
+      Assert.assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_OK);
```
Comment (GrantPSpencer) [inline helix-rest/src/test/java/org/apache/helix/rest/client/TestCustomRestClient.java:164]: Is this a redundant assertion? Above if statement already checks 
`response.getStatusLine().getStatusCode() == HttpStatus.SC_OK`

Comment (desaikomal) [reply helix-rest/src/test/java/org/apache/helix/rest/client/TestCustomRestClient.java:164]: true, it is redundant. I was wondering if there is a better way to detect if the endpoint is accessible? i couldn't find any other way. do you have any suggestion?

File: helix-rest/src/test/java/org/apache/helix/rest/client/TestCustomRestClient.java [hunk 2]
```diff
@@ -158,11 +158,12 @@ public void testPostRequestFormat() throws IOException {
     final String echoServer = "http://httpbin.org/post";
     CustomRestClientImpl customRestClient = new CustomRestClientImpl(HttpClients.createDefault());
     HttpResponse response = customRestClient.post(echoServer, Collections.emptyMap());
-    JsonNode json = customRestClient.getJsonObject(response);
+    if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
```
Comment (rahulrane50) [inline helix-rest/src/test/java/org/apache/helix/rest/client/TestCustomRestClient.java:161]: Shouldn't we assert in else block as well? Or we are expecting any other status code is fine?

Comment (desaikomal) [reply helix-rest/src/test/java/org/apache/helix/rest/client/TestCustomRestClient.java:161]: Thanks for review. Test is ECHOing the request to outside server. If you can't reach the outside server, test used to fail. That is why we validate only if we can successfully reach the end-point. Else part, we can't validate.

**Comment by desaikomal** [comment]
Thanks @GrantPSpencer, @rahulrane50 and @xyuanlu for reviewing the change. This change has been approved by @xyuanlu and ready to merge.
Commit message: Test is trying to reach endpoint echobin.org which MAY not be available. So test only when the site is healthy.

**Pull request merged**